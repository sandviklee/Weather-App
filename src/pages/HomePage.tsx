import { Link } from "react-router-dom";
import { useState, useEffect } from "react";
import styles from "./../style/Home.module.css";
import Card from "../components/card/Card";
import Field from "../components/input/field/Field";
import WeatherIcon from "../components/icon/WeatherIcon";
import Icon from "../components/icon/Icon";
import Selector from "../components/input/selector/Selector";
import { UseQueryResult, useQuery } from "@tanstack/react-query";

interface PlaceSearchInterface {
  name: string;
  municipality: string;
  county: string;
}

const PlaceSearch = ({ name, municipality, county }: PlaceSearchInterface) => {
  function capitalizeWords(input: string) {
    return input
      .split(" ") // Split by spaces
      .map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()) // Capitalize first letter and make the rest lowercase
      .join(" "); // Join back together
  }

  function unsamify(input: string) {
    return input.split("-")[0].trim(); // Split by "-" and take the first part, then trim any whitespace
  }

  return (
    <div className={styles.place_search}>
      <div className={styles.place}>
        <Icon icon="map-pin" size={20} />
        <p>{capitalizeWords(name)}</p>
      </div>
      <div className={styles.extra}>
        <p>{unsamify(municipality)},</p>
        <p>{unsamify(county)}</p>
      </div>
    </div>
  );
};

const HomePage = () => {
  const [currentDate, setCurrentDate] = useState(new Date());
  const [fieldValue, setFieldValue] = useState<string>("");

  useEffect(() => {
    const interval = setInterval(() => {
      setCurrentDate(new Date());
    }, 10000); // Update every second

    // Cleanup: clear the interval when the component is unmounted
    return () => clearInterval(interval);
  }, []);

  const timeOptions: Intl.DateTimeFormatOptions = {
    hour: "2-digit",
    minute: "2-digit",
    hour12: false, // 24-hour clock
  };
  const formattedTime = currentDate.toLocaleTimeString(undefined, timeOptions);

  const dagNavn = [
    "Søndag",
    "Mandag",
    "Tirsdag",
    "Onsdag",
    "Torsdag",
    "Fredag",
    "Lørdag",
  ];

  type Plass = {
    stedsnavn: [{ skrivemåte: string }];
    representasjonspunkt: {
      øst: number;
      nord: number;
      koordsys: 4258;
    };
    fylker: [
      {
        fylkesnavn: string;
        fylkesnummer: number;
      },
    ];
    kommuner: [
      {
        kommunenummer: string;
        kommunenavn: string;
      },
    ];
    stedsnummer: number;
  };

  const dag = dagNavn[currentDate.getDay()];

  const favorites_test = ["Trondheim", "Oslo", "Fornebu", "Bergen"];

  localStorage.setItem("favorites", JSON.stringify(favorites_test));

  const favorites = JSON.parse(localStorage.getItem("favorites") || "[]");

  const [currentLocation, setCurrentLocation] = useState<string | null>(
    favorites[0]
  );

  const { data, refetch } = useQuery({
    /**
     * @summary TanStack Query, fetches API from MET to data
     */
    queryKey: ["search", fieldValue],
    queryFn: () =>
      fetch(
        `https://ws.geonorge.no/stedsnavn/v1/sted?sok=${fieldValue}&fuzzy=true&utkoordsys=4258&treffPerSide=20&side=1`
      )
        .then((res) => res.json())
        .then((data) => {
          const resultsArray = data.navn; // Extracting the array from data.navn

          if (!Array.isArray(resultsArray)) {
            throw new Error("Expected an array in the API response");
          }

          // Filter out results where kommuner, fylker, or stedsnavn are empty or not defined
          const filteredData = resultsArray.filter((item: Plass) => {
            const hasKommuner = item.kommuner && item.kommuner.length > 0;
            const hasFylker = item.fylker && item.fylker.length > 0;
            const hasStedsnavn = item.stedsnavn && item.stedsnavn.length > 0;

            return hasKommuner && hasFylker && hasStedsnavn;
          });

          // Filter out duplicates based on the combination
          const uniqueData = filteredData.reduce(
            (acc: Plass[], curr: Plass) => {
              const uniqueKey = `${curr.fylker[0].fylkesnavn}-${curr.kommuner[0].kommunenummer}-${curr.stedsnavn[0].skrivemåte}`;
              if (
                !acc.some(
                  (item: Plass) =>
                    `${item.fylker[0].fylkesnavn}-${item.kommuner[0].kommunenummer}-${item.stedsnavn[0].skrivemåte}` ===
                    uniqueKey
                )
              ) {
                acc.push(curr);
              }
              return acc;
            },
            []
          );

          return uniqueData;
        }),
  });

  useEffect(() => {
    refetch();
    console.log(data);
    console.log(fieldValue);
  }, [fieldValue, refetch]);

  return (
    <>
      {fieldValue.length > 0 && (
        <div className={styles.search} onClick={() => setFieldValue("")}>
          <div className={styles.search_box}></div>
        </div>
      )}
      <div
        className={styles.input}
        style={
          fieldValue.length > 0
            ? {
                top: "200px",
                width: "80%",
                left: "10%",
              }
            : { padding: "0px" }
        }
      >
        <Field
          placeholder="Søk for områder"
          icon={"search"}
          value={fieldValue}
          setValue={setFieldValue}
        />
        {fieldValue.length > 0 && (
          <>
            <div className={styles.divider}></div>
            <div className={styles.places}>
              {data &&
                data.map((plass: Plass) => (
                  <PlaceSearch
                    name={
                      plass.fylker?.length > 0
                        ? plass.stedsnavn[0].skrivemåte
                        : "Ukjent"
                    }
                    municipality={
                      plass.kommuner?.length > 0
                        ? plass.kommuner[0].kommunenavn
                        : "Ukjent"
                    }
                    county={
                      plass.fylker?.length > 0
                        ? plass.fylker[0].fylkesnavn
                        : "Ukjent"
                    }
                  />
                ))}
            </div>
          </>
        )}
      </div>
      <main className={styles.main}>
        <div className={styles.sidebar}>
          <div className={styles.current_weather}>
            <WeatherIcon dayOrNight="day" status="sunny" size={120} />
          </div>
          <h3 className={styles.temperature}>15°C</h3>
          <h3 className={styles.location}>{currentLocation}</h3>
          <h3 className={styles.day_time}>
            <span className={styles.day}>{dag},</span>{" "}
            <span className={styles.time}>{formattedTime}</span>
          </h3>
          <div className={styles.extra_info}>
            <WeatherIcon dayOrNight="day" status="sunny" size={40} />
            <h4>Sol</h4>
          </div>
          <div className={styles.extra_info}>
            <WeatherIcon dayOrNight="neutral" status="rain" size={40} />
            <h4>Nedbør - 10mm</h4>
          </div>
          <div className={styles.link_container}>
            <Link to="/Weather/Trondheim" className={styles.link}>
              <p>Trykk her for å se mer info for {currentLocation}</p>
              <Icon icon="arrow-right" size={25} />
            </Link>
          </div>
        </div>
        <div className={styles.favorites}>
          <h3>Oversikt over dine plasser</h3>
          <Selector selections={["I dag", "I morgen"]} />
          <div className={styles.card_container}>
            {favorites.map((location: string, index: number) => (
              <button
                onClick={() => setCurrentLocation(location)}
                className={styles.button}
              >
                <Card
                  key={index}
                  location={location}
                  temperature={15}
                  selected={location == currentLocation}
                  nightTemperature={-3}
                />
              </button>
            ))}
          </div>
          <div className={styles.header}>
            <h3>
              Høydepunkter for{" "}
              <span style={{ textDecoration: "underline" }}>
                {currentLocation}
              </span>
            </h3>
            <Link to="/Weather/Trondheim" className={styles.link2}>
              <div className={styles.link2}>
                <p>Se mer info</p>
                <Icon icon="arrow-right" size={25} />
              </div>
            </Link>
          </div>
        </div>
      </main>
    </>
  );
};

export default HomePage;
