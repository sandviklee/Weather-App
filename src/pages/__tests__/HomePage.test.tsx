import { render, screen } from '@testing-library/react';
import HomePage from '../HomePage';
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { BrowserRouter as Router } from 'react-router-dom';
import {test , describe , expect, vi} from "vitest";
import { userEvent } from '@testing-library/user-event';

// Create a client for React Query
const queryClient = new QueryClient();

describe('HomePage component', () => {
    test('renders correctly with 0 favorites', () => {
        render(
            <Router>
                <QueryClientProvider client={queryClient}>
                    <HomePage />
                </QueryClientProvider>
            </Router>
        );

        // Check if the search field is rendered
        const searchField = screen.getByPlaceholderText('Søk for områder');
        expect(searchField).toBeInTheDocument();

        // Check if the no places message is rendered when there are no favorites
        const noPlacesMessage = screen.getByText('Du har ikke lagt til noen lokasjoner enda, søk på en plass for å komme i gang!');
        expect(noPlacesMessage).toBeInTheDocument();
    });

    test('renders favorites correctly', async () => {   
            // Mock favorites data
        const favorites = [
            { name: 'Favorite 1', lat: 123.456, lon: 789.012 },
            { name: 'Favorite 2', lat: 345.678, lon: 901.234 },
            { name: 'Favorite 3', lat: 567.890, lon: 123.456 },
        ];
        
        // Mock localStorage to return the favorites data
        vi.spyOn(window.localStorage.__proto__, 'getItem');
        window.localStorage.__proto__.getItem = vi.fn(() => JSON.stringify(favorites));
        render(
            <QueryClientProvider client={queryClient}>
                <Router>
                    <HomePage />
                </Router>
            </QueryClientProvider>
        );
    
        // Check if the favorites are rendered
        const favorite1 = await screen.findAllByText('Favorite 1');
        // string "Favorite 1" will be found twice, once in the favorites list and once in the side bar 
        // since it is the first favorite and therefore selected by default
        expect(favorite1).toHaveLength(2);
        const favorite2 = await screen.findByText('Favorite 2');
        expect(favorite2).toBeInTheDocument();
        const favorite3 = await screen.findByText('Favorite 3');
        expect(favorite3).toBeInTheDocument();
    });

    test("Favorite 2 is selected", async () => {
            // Mock favorites data
        const favorites = [
            { name: 'Favorite 1', lat: 123.456, lon: 789.012 },
            { name: 'Favorite 2', lat: 345.678, lon: 901.234 },
            { name: 'Favorite 3', lat: 567.890, lon: 123.456 },
        ];
        
        // Mock localStorage to return the favorites data
        vi.spyOn(window.localStorage.__proto__, 'getItem');
        window.localStorage.__proto__.getItem = vi.fn(() => JSON.stringify(favorites));
        render(
            <QueryClientProvider client={queryClient}>
                <Router>
                    <HomePage />
                </Router>
            </QueryClientProvider>
        );

        // Check if Favorite 2 is selected
        const favorite2 = await screen.findByText('Favorite 2');
        expect(favorite2).toBeInTheDocument();
        favorite2.click();

        const newFavorite2 = await screen.findByText('Favorite 2');
        expect(newFavorite2.parentElement).toHaveStyle({
            backgroundColor: "rgb(0, 0, 0)",
            color: "rgb(255, 255, 255)",
        });
    });

    test("matches snapshot with no favorites", () => {
        const { asFragment } = render(
            <QueryClientProvider client={queryClient}>
                <Router>
                    <HomePage />
                </Router>
            </QueryClientProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test("matches snapshot with favorites", () => {
            // Mock favorites data
        const favorites = [
            { name: 'Favorite 1', lat: 123.456, lon: 789.012 },
            { name: 'Favorite 2', lat: 345.678, lon: 901.234 },
            { name: 'Favorite 3', lat: 567.890, lon: 123.456 },
        ];
        
        // Mock localStorage to return the favorites data
        vi.spyOn(window.localStorage.__proto__, 'getItem');
        window.localStorage.__proto__.getItem = vi.fn(() => JSON.stringify(favorites));
        const { asFragment } = render(
            <QueryClientProvider client={queryClient}>
                <Router>
                    <HomePage />
                </Router>
            </QueryClientProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
    
    test("Search renders correctly", async () => {
        render(
            <QueryClientProvider client={queryClient}>
                <Router>
                    <HomePage />
                </Router>
            </QueryClientProvider>
        );
        const searchField = screen.getByPlaceholderText('Søk for områder');
        expect(searchField).toBeInTheDocument();

        // triger search popup
        await userEvent.type(searchField, "lol");
        
        // the test string only appears in the search popup therefore we can use it to check if the popup is rendered
        const filter = await screen.getByText("Tilbakestill filtre");
        expect(filter).toBeInTheDocument();
    });

    test("Search matches snapshot", async () => {
        const { asFragment } = render(
            <QueryClientProvider client={queryClient}>
                <Router>
                    <HomePage />
                </Router>
            </QueryClientProvider>
        );
        const searchField = screen.getByPlaceholderText('Søk for områder');

        // triger search popup
        await userEvent.type(searchField, "lol");
        
        await expect(asFragment()).toMatchSnapshot();
    });
});