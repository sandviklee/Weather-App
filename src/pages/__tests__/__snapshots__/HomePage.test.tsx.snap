// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage component > Search matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_search_6a3077"
  >
    <div
      class="_search_box_6a3077"
    />
  </div>
  <div
    class="_input_6a3077"
    style="padding: 0px 0px 0px 1vw; top: 10vw; width: 80%; left: 8%;"
  >
    <div
      class="_field_47a843"
    >
      <img
        alt="WeatherStatus"
        draggable="false"
        height="20"
        src="/project1/icons/search.svg"
        style="margin: 0px;"
        width="20"
      />
      <input
        class="_input_47a843"
        placeholder="Søk for områder"
        type="text"
        value="lol"
      />
    </div>
    <div
      class="_options_6a3077"
    >
      <div>
        <select
          class="_select_4532b2"
        >
          <option
            disabled=""
            value=""
          >
            Fylke
          </option>
          <option
            value="Troms og Finnmark"
          >
            Troms og Finnmark
          </option>
          <option
            value="Nordland"
          >
            Nordland
          </option>
          <option
            value="Trøndelag"
          >
            Trøndelag
          </option>
          <option
            value="Møre og Romsdal"
          >
            Møre og Romsdal
          </option>
          <option
            value="Vestland"
          >
            Vestland
          </option>
          <option
            value="Rogaland"
          >
            Rogaland
          </option>
          <option
            value="Agder"
          >
            Agder
          </option>
          <option
            value="Vestfold og Telemark"
          >
            Vestfold og Telemark
          </option>
          <option
            value="Viken"
          >
            Viken
          </option>
          <option
            value="Oslo"
          >
            Oslo
          </option>
          <option
            value="Innlandet"
          >
            Innlandet
          </option>
        </select>
      </div>
      <p
        class="_reset_6a3077"
      >
        Tilbakestill filtre
      </p>
    </div>
    <div
      class="_divider_6a3077"
    />
    <div
      class="_places_6a3077"
    />
  </div>
  <main
    class="_main_6a3077"
  >
    <div
      class="_sidebar_6a3077"
    >
      <div
        class="_current_weather_6a3077"
      >
        <img
          alt="WeatherStatus"
          draggable="false"
          height="120"
          src="/project1/yr-icons/undefined.svg"
          style="margin: 0px;"
          width="120"
        />
      </div>
      <h3
        class="_temperature_6a3077"
      >
        °C
      </h3>
      <h3
        class="_location_6a3077"
      >
        Favorite 1
      </h3>
      <h3
        class="_day_time_6a3077"
      >
        <span
          class="_day_6a3077"
        >
          Søndag,
        </span>
         
        <span
          class="_time_6a3077"
        >
          15:53
        </span>
      </h3>
      <div
        class="_extra_info_6a3077"
      >
        <img
          alt="WeatherStatus"
          draggable="false"
          height="30"
          src="/project1/yr-icons/undefined.svg"
          style="margin: 0px;"
          width="30"
        />
        <h4 />
      </div>
      <div
        class="_extra_info_6a3077"
      >
        <img
          alt="WeatherStatus"
          draggable="false"
          height="30"
          src="/project1/yr-icons/rain.svg"
          style="margin: 0px;"
          width="30"
        />
        <h4>
          Nedbør - mm
        </h4>
      </div>
    </div>
    <div
      class="_favorites_6a3077"
    >
      <div
        class="_header_6a3077"
      >
        <p
          class="_container_title_6a3077"
        >
          Dine plasser
        </p>
        <a
          class="_link_6a3077"
          href="/location/Favorite 1/123.456/789.012"
        >
          <p>
            Se mer info for Favorite 1
          </p>
          <img
            alt="WeatherStatus"
            draggable="false"
            height="25"
            src="/project1/icons/arrow-right.svg"
            style="margin: 0px;"
            width="25"
          />
        </a>
      </div>
      <div
        class="_card_container_6a3077"
      >
        <button
          class="_button_6a3077"
        >
          <div
            class="_card_3c4546"
          >
            <div
              class="_container_3c4546"
              style="background-color: black; color: white;"
            >
              <h1
                class="_location_3c4546"
              >
                Favorite 1
              </h1>
              <img
                alt="WeatherStatus"
                draggable="false"
                height="24"
                src="/project1/yr-icons/clearsky_day.svg"
                style="margin: 0px;"
                width="24"
              />
              <div
                class="_temperatures_3c4546"
              >
                <p
                  class="_temperature_3c4546"
                >
                  0°
                </p>
                <p
                  class="_night_temperature_3c4546"
                >
                  0°
                </p>
              </div>
            </div>
          </div>
        </button>
        <button
          class="_button_6a3077"
        >
          <div
            class="_card_3c4546"
          >
            <div
              class="_container_3c4546"
              style="background-color: white;"
            >
              <h1
                class="_location_3c4546"
              >
                Favorite 2
              </h1>
              <img
                alt="WeatherStatus"
                draggable="false"
                height="24"
                src="/project1/yr-icons/clearsky_day.svg"
                style="margin: 0px;"
                width="24"
              />
              <div
                class="_temperatures_3c4546"
              >
                <p
                  class="_temperature_3c4546"
                >
                  0°
                </p>
                <p
                  class="_night_temperature_3c4546"
                >
                  0°
                </p>
              </div>
            </div>
          </div>
        </button>
        <button
          class="_button_6a3077"
        >
          <div
            class="_card_3c4546"
          >
            <div
              class="_container_3c4546"
              style="background-color: white;"
            >
              <h1
                class="_location_3c4546"
              >
                Favorite 3
              </h1>
              <img
                alt="WeatherStatus"
                draggable="false"
                height="24"
                src="/project1/yr-icons/clearsky_day.svg"
                style="margin: 0px;"
                width="24"
              />
              <div
                class="_temperatures_3c4546"
              >
                <p
                  class="_temperature_3c4546"
                >
                  0°
                </p>
                <p
                  class="_night_temperature_3c4546"
                >
                  0°
                </p>
              </div>
            </div>
          </div>
        </button>
      </div>
      <h3
        class="_title_6a3077"
      >
        Høydepunkter for Favorite 1
      </h3>
      <div
        class="_card_container_6a3077"
      >
        <div
          class="_container_a120ba"
        >
          <p
            class="_title_a120ba"
          >
            UV Index
          </p>
          <div
            class="_bar_container_a120ba"
          >
            <div
              class="_bar_a120ba"
            >
              <div
                style="background-color: rgb(255, 142, 60); height: 100%;"
              />
            </div>
            <p
              class="_uv_index_a120ba"
            />
          </div>
        </div>
        <div
          class="_container_1d4056"
        >
          <p
            class="_title_1d4056"
          >
            Vind
          </p>
          <div
            class="_info_container_1d4056"
          >
            <p
              class="_uv_index_1d4056"
            >
              m/s 
              <span
                style="color: rgba(60, 103, 255, 0.6);"
              >
                 ()
              </span>
            </p>
            <div
              class="_compass_1d4056"
            >
              <img
                alt="arrow"
                height="24px"
                src="/project1/icons/compass.svg"
                style="transform: rotate(undefineddeg);"
                width="24px"
              />
            </div>
          </div>
        </div>
        <div
          class="_container_7ae00b"
        >
          <p
            class="_title_7ae00b"
          >
            Luftfuktighet
          </p>
          <div
            class="_humidity_container_7ae00b"
          >
            <div
              class="_status_container_7ae00b"
            >
              <div
                class="_circle_7ae00b"
                style="background-color: rgb(255, 141, 141);"
              />
              <p>
                Høy
              </p>
            </div>
            <p
              class="_humidity_7ae00b"
            >
              NaN%
            </p>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`HomePage component > matches snapshot with favorites 1`] = `
<DocumentFragment>
  <div
    class="_input_6a3077"
    style="padding: 0px 0px 0px 1vw;"
  >
    <div
      class="_field_47a843"
    >
      <img
        alt="WeatherStatus"
        draggable="false"
        height="20"
        src="/project1/icons/search.svg"
        style="margin: 0px;"
        width="20"
      />
      <input
        class="_input_47a843"
        placeholder="Søk for områder"
        type="text"
        value=""
      />
    </div>
  </div>
  <main
    class="_main_6a3077"
  >
    <div
      class="_sidebar_6a3077"
    >
      <div
        class="_current_weather_6a3077"
      >
        <img
          alt="WeatherStatus"
          draggable="false"
          height="120"
          src="/project1/yr-icons/undefined.svg"
          style="margin: 0px;"
          width="120"
        />
      </div>
      <h3
        class="_temperature_6a3077"
      >
        °C
      </h3>
      <h3
        class="_location_6a3077"
      >
        Favorite 1
      </h3>
      <h3
        class="_day_time_6a3077"
      >
        <span
          class="_day_6a3077"
        >
          Søndag,
        </span>
         
        <span
          class="_time_6a3077"
        >
          15:53
        </span>
      </h3>
      <div
        class="_extra_info_6a3077"
      >
        <img
          alt="WeatherStatus"
          draggable="false"
          height="30"
          src="/project1/yr-icons/undefined.svg"
          style="margin: 0px;"
          width="30"
        />
        <h4 />
      </div>
      <div
        class="_extra_info_6a3077"
      >
        <img
          alt="WeatherStatus"
          draggable="false"
          height="30"
          src="/project1/yr-icons/rain.svg"
          style="margin: 0px;"
          width="30"
        />
        <h4>
          Nedbør - mm
        </h4>
      </div>
    </div>
    <div
      class="_favorites_6a3077"
    >
      <div
        class="_header_6a3077"
      >
        <p
          class="_container_title_6a3077"
        >
          Dine plasser
        </p>
        <a
          class="_link_6a3077"
          href="/location/Favorite 1/123.456/789.012"
        >
          <p>
            Se mer info for Favorite 1
          </p>
          <img
            alt="WeatherStatus"
            draggable="false"
            height="25"
            src="/project1/icons/arrow-right.svg"
            style="margin: 0px;"
            width="25"
          />
        </a>
      </div>
      <div
        class="_card_container_6a3077"
      >
        <button
          class="_button_6a3077"
        >
          <div
            class="_card_3c4546"
          >
            <div
              class="_container_3c4546"
              style="background-color: black; color: white;"
            >
              <h1
                class="_location_3c4546"
              >
                Favorite 1
              </h1>
              <img
                alt="WeatherStatus"
                draggable="false"
                height="24"
                src="/project1/yr-icons/clearsky_day.svg"
                style="margin: 0px;"
                width="24"
              />
              <div
                class="_temperatures_3c4546"
              >
                <p
                  class="_temperature_3c4546"
                >
                  0°
                </p>
                <p
                  class="_night_temperature_3c4546"
                >
                  0°
                </p>
              </div>
            </div>
          </div>
        </button>
        <button
          class="_button_6a3077"
        >
          <div
            class="_card_3c4546"
          >
            <div
              class="_container_3c4546"
              style="background-color: white;"
            >
              <h1
                class="_location_3c4546"
              >
                Favorite 2
              </h1>
              <img
                alt="WeatherStatus"
                draggable="false"
                height="24"
                src="/project1/yr-icons/clearsky_day.svg"
                style="margin: 0px;"
                width="24"
              />
              <div
                class="_temperatures_3c4546"
              >
                <p
                  class="_temperature_3c4546"
                >
                  0°
                </p>
                <p
                  class="_night_temperature_3c4546"
                >
                  0°
                </p>
              </div>
            </div>
          </div>
        </button>
        <button
          class="_button_6a3077"
        >
          <div
            class="_card_3c4546"
          >
            <div
              class="_container_3c4546"
              style="background-color: white;"
            >
              <h1
                class="_location_3c4546"
              >
                Favorite 3
              </h1>
              <img
                alt="WeatherStatus"
                draggable="false"
                height="24"
                src="/project1/yr-icons/clearsky_day.svg"
                style="margin: 0px;"
                width="24"
              />
              <div
                class="_temperatures_3c4546"
              >
                <p
                  class="_temperature_3c4546"
                >
                  0°
                </p>
                <p
                  class="_night_temperature_3c4546"
                >
                  0°
                </p>
              </div>
            </div>
          </div>
        </button>
      </div>
      <h3
        class="_title_6a3077"
      >
        Høydepunkter for Favorite 1
      </h3>
      <div
        class="_card_container_6a3077"
      >
        <div
          class="_container_a120ba"
        >
          <p
            class="_title_a120ba"
          >
            UV Index
          </p>
          <div
            class="_bar_container_a120ba"
          >
            <div
              class="_bar_a120ba"
            >
              <div
                style="background-color: rgb(255, 142, 60); height: 100%;"
              />
            </div>
            <p
              class="_uv_index_a120ba"
            />
          </div>
        </div>
        <div
          class="_container_1d4056"
        >
          <p
            class="_title_1d4056"
          >
            Vind
          </p>
          <div
            class="_info_container_1d4056"
          >
            <p
              class="_uv_index_1d4056"
            >
              m/s 
              <span
                style="color: rgba(60, 103, 255, 0.6);"
              >
                 ()
              </span>
            </p>
            <div
              class="_compass_1d4056"
            >
              <img
                alt="arrow"
                height="24px"
                src="/project1/icons/compass.svg"
                style="transform: rotate(undefineddeg);"
                width="24px"
              />
            </div>
          </div>
        </div>
        <div
          class="_container_7ae00b"
        >
          <p
            class="_title_7ae00b"
          >
            Luftfuktighet
          </p>
          <div
            class="_humidity_container_7ae00b"
          >
            <div
              class="_status_container_7ae00b"
            >
              <div
                class="_circle_7ae00b"
                style="background-color: rgb(255, 141, 141);"
              />
              <p>
                Høy
              </p>
            </div>
            <p
              class="_humidity_7ae00b"
            >
              NaN%
            </p>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`HomePage component > matches snapshot with no favorites 1`] = `
<DocumentFragment>
  <div
    class="_input_6a3077"
    style="padding: 0px 0px 0px 1vw;"
  >
    <div
      class="_field_47a843"
    >
      <img
        alt="WeatherStatus"
        draggable="false"
        height="20"
        src="/project1/icons/search.svg"
        style="margin: 0px;"
        width="20"
      />
      <input
        class="_input_47a843"
        placeholder="Søk for områder"
        type="text"
        value=""
      />
    </div>
  </div>
  <main
    class="_main_6a3077"
  >
    <div
      class="_sidebar_6a3077"
    >
      <div
        class="_current_weather_6a3077"
      >
        <img
          alt="WeatherStatus"
          draggable="false"
          height="120"
          src="/project1/yr-icons/undefined.svg"
          style="margin: 0px;"
          width="120"
        />
      </div>
      <h3
        class="_temperature_6a3077"
      >
        °C
      </h3>
      <h3
        class="_location_6a3077"
      >
        Favorite 1
      </h3>
      <h3
        class="_day_time_6a3077"
      >
        <span
          class="_day_6a3077"
        >
          Søndag,
        </span>
         
        <span
          class="_time_6a3077"
        >
          15:53
        </span>
      </h3>
      <div
        class="_extra_info_6a3077"
      >
        <img
          alt="WeatherStatus"
          draggable="false"
          height="30"
          src="/project1/yr-icons/undefined.svg"
          style="margin: 0px;"
          width="30"
        />
        <h4 />
      </div>
      <div
        class="_extra_info_6a3077"
      >
        <img
          alt="WeatherStatus"
          draggable="false"
          height="30"
          src="/project1/yr-icons/rain.svg"
          style="margin: 0px;"
          width="30"
        />
        <h4>
          Nedbør - mm
        </h4>
      </div>
    </div>
    <div
      class="_favorites_6a3077"
    >
      <div
        class="_header_6a3077"
      >
        <p
          class="_container_title_6a3077"
        >
          Dine plasser
        </p>
        <a
          class="_link_6a3077"
          href="/location/Favorite 1/123.456/789.012"
        >
          <p>
            Se mer info for Favorite 1
          </p>
          <img
            alt="WeatherStatus"
            draggable="false"
            height="25"
            src="/project1/icons/arrow-right.svg"
            style="margin: 0px;"
            width="25"
          />
        </a>
      </div>
      <div
        class="_card_container_6a3077"
      >
        <button
          class="_button_6a3077"
        >
          <div
            class="_card_3c4546"
          >
            <div
              class="_container_3c4546"
              style="background-color: black; color: white;"
            >
              <h1
                class="_location_3c4546"
              >
                Favorite 1
              </h1>
              <img
                alt="WeatherStatus"
                draggable="false"
                height="24"
                src="/project1/yr-icons/clearsky_day.svg"
                style="margin: 0px;"
                width="24"
              />
              <div
                class="_temperatures_3c4546"
              >
                <p
                  class="_temperature_3c4546"
                >
                  0°
                </p>
                <p
                  class="_night_temperature_3c4546"
                >
                  0°
                </p>
              </div>
            </div>
          </div>
        </button>
        <button
          class="_button_6a3077"
        >
          <div
            class="_card_3c4546"
          >
            <div
              class="_container_3c4546"
              style="background-color: white;"
            >
              <h1
                class="_location_3c4546"
              >
                Favorite 2
              </h1>
              <img
                alt="WeatherStatus"
                draggable="false"
                height="24"
                src="/project1/yr-icons/clearsky_day.svg"
                style="margin: 0px;"
                width="24"
              />
              <div
                class="_temperatures_3c4546"
              >
                <p
                  class="_temperature_3c4546"
                >
                  0°
                </p>
                <p
                  class="_night_temperature_3c4546"
                >
                  0°
                </p>
              </div>
            </div>
          </div>
        </button>
        <button
          class="_button_6a3077"
        >
          <div
            class="_card_3c4546"
          >
            <div
              class="_container_3c4546"
              style="background-color: white;"
            >
              <h1
                class="_location_3c4546"
              >
                Favorite 3
              </h1>
              <img
                alt="WeatherStatus"
                draggable="false"
                height="24"
                src="/project1/yr-icons/clearsky_day.svg"
                style="margin: 0px;"
                width="24"
              />
              <div
                class="_temperatures_3c4546"
              >
                <p
                  class="_temperature_3c4546"
                >
                  0°
                </p>
                <p
                  class="_night_temperature_3c4546"
                >
                  0°
                </p>
              </div>
            </div>
          </div>
        </button>
      </div>
      <h3
        class="_title_6a3077"
      >
        Høydepunkter for Favorite 1
      </h3>
      <div
        class="_card_container_6a3077"
      >
        <div
          class="_container_a120ba"
        >
          <p
            class="_title_a120ba"
          >
            UV Index
          </p>
          <div
            class="_bar_container_a120ba"
          >
            <div
              class="_bar_a120ba"
            >
              <div
                style="background-color: rgb(255, 142, 60); height: 100%;"
              />
            </div>
            <p
              class="_uv_index_a120ba"
            />
          </div>
        </div>
        <div
          class="_container_1d4056"
        >
          <p
            class="_title_1d4056"
          >
            Vind
          </p>
          <div
            class="_info_container_1d4056"
          >
            <p
              class="_uv_index_1d4056"
            >
              m/s 
              <span
                style="color: rgba(60, 103, 255, 0.6);"
              >
                 ()
              </span>
            </p>
            <div
              class="_compass_1d4056"
            >
              <img
                alt="arrow"
                height="24px"
                src="/project1/icons/compass.svg"
                style="transform: rotate(undefineddeg);"
                width="24px"
              />
            </div>
          </div>
        </div>
        <div
          class="_container_7ae00b"
        >
          <p
            class="_title_7ae00b"
          >
            Luftfuktighet
          </p>
          <div
            class="_humidity_container_7ae00b"
          >
            <div
              class="_status_container_7ae00b"
            >
              <div
                class="_circle_7ae00b"
                style="background-color: rgb(255, 141, 141);"
              />
              <p>
                Høy
              </p>
            </div>
            <p
              class="_humidity_7ae00b"
            >
              NaN%
            </p>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
